cmd/scality-cosi-driver/cmd.go

```diff
@@ -25,27 +25,39 @@ import (
 	"github.com/scality/cosi-driver/pkg/metrics"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	stdout "go.opentelemetry.io/otel/exporters/stdout/stdouttrace"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"k8s.io/klog/v2"
 )
 
 const (
-	provisionerName       = "scality.com"
-	defaultDriverAddress  = "unix:///var/lib/cosi/cosi.sock"
-	defaultDriverPrefix   = "cosi"
-	defaultMetricsPath    = "/metrics"
-	defaultMetricsPrefix  = "scality_cosi_driver"
-	defaultMetricsAddress = ":8080"
+	provisionerName        = "scality.com"
+	defaultDriverAddress   = "unix:///var/lib/cosi/cosi.sock"
+	defaultDriverPrefix    = "cosi"
+	defaultMetricsPath     = "/metrics"
+	defaultMetricsPrefix   = "scality_cosi_driver"
+	defaultMetricsAddress  = ":8080"
+	defaultOtelStdout      = false
+	defaultOtelEndpoint    = "localhost:4318"
+	defaultOtelServiceName = "cosi.scality.com"
 )
 
 var (
-	driverAddress        = flag.String("driver-address", defaultDriverAddress, "driver address for the socket file, default: unix:///var/lib/cosi/cosi.sock")
-	driverPrefix         = flag.String("driver-prefix", defaultDriverPrefix, "prefix for COSI driver, e.g. <prefix>.scality.com, default cosi.scality.com")
-	driverMetricsAddress = flag.String("driver-metrics-address", defaultMetricsAddress, "The address to expose Prometheus metrics, default: :8080")
-	driverMetricsPath    = flag.String("driver-metrics-path", defaultMetricsPath, "path for the metrics endpoint, default: /metrics")
-	driverMetricsPrefix  = flag.String("driver-custom-metrics-prefix", defaultMetricsPrefix, "prefix for the metrics, default: scality_cosi_driver_")
+	driverAddress         = flag.String("driver-address", defaultDriverAddress, "Driver address for the socket file, default: unix:///var/lib/cosi/cosi.sock")
+	driverPrefix          = flag.String("driver-prefix", defaultDriverPrefix, "Prefix for COSI driver, e.g. <prefix>.scality.com, default: cosi.scality.com")
+	driverMetricsAddress  = flag.String("driver-metrics-address", defaultMetricsAddress, "The address to expose Prometheus metrics, default: :8080")
+	driverMetricsPath     = flag.String("driver-metrics-path", defaultMetricsPath, "Path for the metrics endpoint, default: /metrics")
+	driverMetricsPrefix   = flag.String("driver-custom-metrics-prefix", defaultMetricsPrefix, "Prefix for the metrics, default: scality_cosi_driver_")
+	driverOtelEndpoint    = flag.String("driver-otel-endpoint", defaultOtelEndpoint, "OpenTelemetry endpoint to export traces, default: localhost:4317")
+	driverOtelStdout      = flag.Bool("driver-otel-stdout", defaultOtelStdout, "Enable OpenTelemetry trace export to stdout, disables endpoint if enabled")
+	driverOtelServiceName = flag.String("driver-otel-service-name", defaultOtelServiceName, "Service name for OpenTelemetry traces")
```

Comment:
- default endpoint is `4318` not `4317`
- missing default in other description
- using `127.0.0.1` instead of `localhost` could be better to avoid any dns resolve ?

Reply:
Changed the part for the port number.
Added defaults.
I do not want to specify the IP address 127.0.0.1 cuz we would never package an Otel exporter / collector with the COSI driver Pod, it needs to be an external service.